File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.HashMap;

class Solution {
    public static String[] solution(String[] players, String[] callings) {
        HashMap<String, Integer> dct = new HashMap<String, Integer>();
        for (int i = 0; i < players.length; i++) {
            dct.put(players[i], i);
        }

        for (String calling : callings) {
            int currentIndex = dct.get(calling);
            int desiredIndex = currentIndex - 1;
            String temp = players[currentIndex];
            players[currentIndex] = players[desiredIndex];
            players[desiredIndex] = temp;
            dct.put(players[currentIndex], currentIndex);
            dct.put(players[desiredIndex], desiredIndex);
        }
        return players;
    }
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    dct = {}
    N = len(players)
    for i in range(N):
        dct[players[i]] = i
    for calling in callings:
        num = dct[calling]
        dct[calling] -= 1
        dct[players[num - 1]] += 1
        players[num], players[num-1] = players[num - 1], players[num]
        
    return players